File: src/eu/insertcode/wordgames/games/CalculateGame.java
package eu.insertcode.wordgames.games;

import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import eu.insertcode.wordgames.Main;

public class CalculateGame extends LongWordGame {
	private static final String PERMISSION_PLAY_TYPE = "permission.play.calculate";
	private static final String PERMISSION_START_TYPE = "permission.start.calculate";
	
	public CalculateGame(Main instance, String wordToType, Reward reward) {
		super(instance, wordToType, reward);
		
		int numberOne = (int) (Math.random() * 51);
		int numberTwo = (int) (Math.random() * 20) + 1;
		switch ((int) (Math.random() * 4)) {
			case 1:
				this.wordToType = Math.round((double) numberOne + numberTwo) + "";
				showedWord = numberOne + " + " + numberTwo;
				break;
			case 2:
				this.wordToType = Math.round((double) numberOne - numberTwo) + "";
				showedWord = numberOne + " - " + numberTwo;
				break;
			case 3:
				this.wordToType = Math.round((double) numberOne * numberTwo) + "";
				showedWord = numberOne + " * " + numberTwo;
				break;
			default:
				this.wordToType = Math.round((double) numberOne / numberTwo) + "";
				showedWord = numberOne + " / " + numberTwo;
		}
		sendGameMessage();
	}
	
	public static boolean hasStartPermission(CommandSender s) {
		return WordGame.hasStartPermission(s) || s.hasPermission(PERMISSION_START_TYPE);
	}
	
	@Override
	public boolean hasPlayPermission(Player p) {
		return super.hasPlayPermission(p) || p.hasPermission(PERMISSION_PLAY_TYPE);
	}
	
	@Override
	String getMessageConfigPath() {
		return "games.calculate";
	}
}

File: src/main/java/eu/insertcode/wordgames/games/CalculateGame.kt
package eu.insertcode.wordgames.games

import eu.insertcode.wordgames.Main
import eu.insertcode.wordgames.Permission
import eu.insertcode.wordgames.config.Messages
import kotlin.math.roundToInt

class CalculateGame(instance: Main, wordToType: String, reward: Reward) : LongWordGame(instance, wordToType, reward) {

    override val playPermission = Permission.PLAY_CALCULATE
    override val messageConfigPath = Messages.Games.calculate

    init {
        val numberOne = (Math.random() * 51).toInt()
        val numberTwo = (Math.random() * 20).toInt() + 1

        when ((Math.random() * 4).toInt()) {
            1 -> {
                this.wordToType = (numberOne + numberTwo).toString()
                showedWord = "$numberOne + $numberTwo"
            }
            2 -> {
                this.wordToType = (numberOne - numberTwo).toString()
                showedWord = "$numberOne - $numberTwo"
            }
            3 -> {
                this.wordToType = (numberOne * numberTwo).toString()
                showedWord = "$numberOne * $numberTwo"
            }
            else -> {
                this.wordToType = (numberOne.toDouble() / numberTwo).roundToInt().toString()
                showedWord = "$numberOne / $numberTwo"
            }
        }

        sendGameMessage()
    }

}